src/plugins/intel_cpu/src/nodes/reshape.cpp

```diff
@@ -130,15 +130,23 @@ void Reshape::execute(dnnl::stream strm) {
     auto srcMemPtr = getSrcMemoryAtPort(0);
     auto dstMemPtr = getDstMemoryAtPort(0);
 
-    auto srcPtr = static_cast<uint8_t*>(srcMemPtr->getData());
-    auto dstPtr = static_cast<uint8_t*>(dstMemPtr->getData());
-
-    if (dstPtr != srcPtr) {
-        cpu_memcpy(dstPtr, srcPtr, dstMemPtr->getSize());
+    if (srcMemPtr->getDesc().getPrecision() == element::string) {
+        auto srcPtr = srcMemPtr->getDataAs<StringMemory::OvString>();
+        auto dstPtr = dstMemPtr->getDataAs<StringMemory::OvString>();
+        std::copy(srcPtr, srcPtr + dstMemPtr->getShape().getElementsCount(), dstPtr);
+    } else {
```

Comment:
Extended accordingly to the CPU changes for string type in reorder node:
https://github.com/openvinotoolkit/openvino/blob/a09033841e475cdd2a7d3ffd9e9bcb25a188f378/src/plugins/intel_cpu/src/nodes/reorder.cpp#L460-L463

Reply:
Reverted to propose changes within a separate PR.